libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/RouterWrapper.kt

```diff
@@ -116,27 +140,51 @@ class RouterWrapper(
         }
     }
 
+    override fun getRoute(routeOptions: RouteOptions, callback: RouterCallback): Long {
+        return getRoute(
+            routeOptions,
+            object : NavigationRouterCallback {
+                override fun onRoutesReady(
+                    routes: List<NavigationRoute>,
+                    routerOrigin: RouterOrigin
+                ) {
+                    callback.onRoutesReady(routes.toDirectionsRoutes(), routerOrigin)
+                }
+
+                override fun onFailure(reasons: List<RouterFailure>, routeOptions: RouteOptions) {
+                    callback.onFailure(reasons, routeOptions)
+                }
+
+                override fun onCanceled(routeOptions: RouteOptions, routerOrigin: RouterOrigin) {
+                    callback.onCanceled(routeOptions, routerOrigin)
+                }
+            }
+        )
+    }
+
+    @OptIn(ExperimentalMapboxNavigationAPI::class)
     override fun getRouteRefresh(
-        route: DirectionsRoute,
+        navigationRoute: NavigationRoute,
         legIndex: Int,
-        callback: RouteRefreshCallback
+        callback: NavigationRouterRefreshCallback
     ): Long {
-        val routeOptions = route.routeOptions()
-        val requestUuid = route.requestUuid()
-        val routeIndex = route.routeIndex()?.toIntOrNull()
-        if (routeOptions == null || requestUuid == null || routeIndex == null) {
+        val routeOptions = navigationRoute.routeOptions
+        val requestUuid = navigationRoute.directionsResponse.uuid()
```

Comment:
Should be something like `NavigationRoute.getRoute().requestUuid()`? Wondering if `DirectionsResponse#uuid` is the same as `DirectionsRoute#requestUuid` (especially if the response has more than one route) 🤔 

Refs. above-mentioned comment.

Reply:
>  Wondering if DirectionsResponse#uuid is the same as DirectionsRoute#requestUuid (especially if the response has more than one route)

It is always the same, especially since `NavigationRoute` practically holds a reference to only one `DirectionsRoute`.